### PR TITLE
[Review] Update handling of ClientConfig

### PR DESCRIPTION
**Pull request by ukaea-chah**
Make a number of related changes intended to improve the handling of UA_ClientConfig and its use with setting up a UA_Client.  Related to #5435.

### CODE REVIEW

File: plugins/ua_config_default.c [hunk 1]
```diff
@@ -823,57 +841,45 @@ UA_ClientConfig_setDefault(UA_ClientConfig *config) {
         config->eventLoop->registerEventSource(config->eventLoop, (UA_EventSource *)udpCM);
     }
 
-    if(config->sessionLocaleIdsSize > 0 && config->sessionLocaleIds) {
```
Comment (ukaea-chah) [inline plugins/ua_config_default.c:835]: It's not clear to me in what circumstances a config with sessionLocaleIds would be presented.  And it seemed heavy-handed to actively delete them in a "setDefault" function.

Comment (jpfr) [reply plugins/ua_config_default.c:835]: True.
We don't need that.

**Comment by jpfr** [comment]
Many CI pipelines fail to run / run out of time.
Please have a look. Otherwise the proposed changes are great.

**Comment by ukaea-chah** [comment]
> Many CI pipelines fail to run / run out of time.

CI now green.  So now I shall squash / rebase.

**Comment by jpfr** [comment]
Perfect PR. 👍

**Pull request merged**